### PR TITLE
Do nothing if the params didn't change when compositing platform views.

### PR DESCRIPTION
**Pull request by cyanglaz**

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h [hunk 1]
```diff
@@ -88,6 +88,7 @@ class FlutterPlatformViewsController {
   std::map<std::string, fml::scoped_nsobject<NSObject<FlutterPlatformViewFactory>>> factories_;
   std::map<int64_t, fml::scoped_nsobject<NSObject<FlutterPlatformView>>> views_;
   std::map<int64_t, fml::scoped_nsobject<FlutterTouchInterceptingView>> touch_interceptors_;
+  std::map<int64_t, EmbeddedViewParams> composite_params_;
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h]: nit: current_composition_params_
and add a comment explaining that it's mapping a platform view ID to its latest composition params.

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm [hunk 2]
```diff
@@ -187,7 +187,13 @@
...
   // TODO(amirh): assert that this is running on the platform thread once we support the iOS
   // embedded views thread configuration.
-  // TODO(amirh): do nothing if the params didn't change.
+
+  // Do nothing if the params didn't change.
+  if (composite_params_.count(view_id) == 1 && composite_params_[view_id] == params) {
+    return picture_recorders_[view_id]->getRecordingCanvas();
+  }
+  composite_params_[view_id] = params;
+
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: Do we need to call `setFrame` after a UIView is removed from the super view and then added again? (if so we should probably clear the value in `DetachUnusedLayers` as well)

File: flow/embedded_views.h [hunk 3]
```diff
@@ -18,6 +18,12 @@ class EmbeddedViewParams {
  public:
   SkPoint offsetPixels;
   SkSize sizePoints;
+
+  friend bool operator==(const EmbeddedViewParams& lhs,
```
Comment (amirh) [inline flow/embedded_views.h]: + @jason-simmons  for C++ style in this part (I've never added a C++ comparison operator in the engine codebase, not sure what to look for)

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm (hunk 2 above)
Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: I am not sure if I follow. What should we clear in `DetachUnusedLayers`?

Comment (amirh) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: I asking if we should clear `current_composition_params[view_id]` when we remove `view_id` from the UIView tree in `DetachUnusedLayer` (e.g if we are re-attaching that view with the same composition parameters, do we need to call setframe again?)

Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: The frame of the subview won't change after removing it from superview. So if we remove the platform view in `DetachUnusedLayer` from the superview and the composite param doesn't change, it is ok to not call setFrame again.

File: flow/embedded_views.h (hunk 3 above)
Comment (jason-simmons) [reply flow/embedded_views.h]: is `friend` necessary here?

Comment (cyanglaz) [reply flow/embedded_views.h]: Done

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h (hunk 1 above)
Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h]: Done

**Pull request merged**